draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -369,6 +370,110 @@ received.
+           <192.0.2.1> |------ srflx Candidate C1.3 ----->|
+      <2001:db8::1     |--Binding Req-->|                 |
+       is 2001:db8::1> |<-Binding Resp--|                 |
+         <2001:db8::1> |------ srflx Candidate C1.4 ----->| <Discard C1.4
+                       |                |                 |  as redundant>
+                       |                |                 |
+      - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+              ICE Agent 2 sends mDNS candidates, resolution is slow
+                       |                |                 |
+                       |<------ mDNS Candidate C2.1 ------| <N2.1>
+                       |<------ mDNS Candidate C2.2 ------| <N2.2>
+       <Resolve mDNS   |                |                 |
+        name N2.1 ...> |                |                 |
+       <Resolve mDNS   |                |                 |
+        name N2.2 ...> |                |                 |
+                       |                |                 |
+      - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+        ICE Agent 2 sends server-reflexive candidates, resolution completes
+                       |                |                 |
+                       |                |<--Binding Req---| <192.168.1.2
+                       |                |---Binding Resp->|  is 192.0.2.2>
+                       |<----- srflx Candidate C2.3 ------| <192.0.2.2>
+                       |                |<--Binding Req---| <2001:db8::2
+                       |                |---Binding Resp->|  is 2001:db8::2>
+                       |<----- srflx Candidate C2.4 ------| <2001:db8::2>
+                       |                |                 |
+        <... N2.1 is   |                |                 |
+         192.168.1.2>  |                |                 |
+        <... N2.2 is   |                |                 |
+         2001:db8::2,  |                |                 |
+         discard C2.4> |                |                 |
+                       |                |                 |
+      - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+                                 ICE processing
+                       |                |                 |
+           2001:db8::1 |<============= STUN ==============| 2001:db8::2
+           2001:db8::1 |============== STUN =============>| 2001:db8::2
+           192.168.1.1 |<============= STUN ==============| 192.168.1.2
+           192.168.1.1 |============== STUN =============>| 192.168.1.2
+           2001:db8::1 |<============= STUN ==============| 2001:db8::2
+           2001:db8::1 |============== STUN =============>| 2001:db8::2
+             192.0.2.1 |    Failed === STUN ==============| 192.168.1.2
+           192.168.1.1 |============== STUN === Failed    | 192.0.2.2
+           2001:db8::1 |====== STUN, USE-CANDIDATE ======>| 2001:db8::2
```

Comment:
This part reads less well than the others. Some comments:

Should we use --- instead of ===?
Do we need two lines for each STUN req/resp? Should we specify STUN request/STUN response?

It seems we have two STUN checks for 2001:db8::1 /2001::db8::2?
Also, since the IPv6 check is successful, do we need to do the other checks?
Maybe we should put 2001:db8::1 as the last one?

The last line is a bit different from the others since it is the selection, maybe it should be spelled out, s/ STUN, USE-CANDIDATE/USE-CANDIDATE/ for instance.

Reply:
> Should we use --- instead of ===?
> Do we need two lines for each STUN req/resp? Should we specify STUN request/STUN response?

The === was intended to indicate that there is a request and a response combined in that line. For a given candidate pair, both sides will send their own requests and receive their own responses. LMK if you think we should split these into their own lines (would need to adjust other samples too). 

> It seems we have two STUN checks for 2001:db8::1 /2001::db8::2?

removed, not sure what I was thinking

> Also, since the IPv6 check is successful, do we need to do the other checks?
> Maybe we should put 2001:db8::1 as the last one?

The ICE agent can use whatever policy it wants on its checks, I think as an illustrative example this is probably OK. If we made the IPv6 check last, that implies it is lower priority and the agent should stop after the successful IPv4 check.

> The last line is a bit different from the others since it is the selection, maybe it should be spelled out, s/ STUN, USE-CANDIDATE/USE-CANDIDATE/ for instance.

Didn't quite follow what you had in mind here. Tweaked the text slightly in an attempt to clarify.

re: duplicate checks; 

re: IPv6 check vs other checks, t

Reply:
ping @youennf

Reply:
ping @youennf